File: ordered_words.rb
#!/usr/bin/env ruby
# http://programmingpraxis.com/
# http://rosettacode.org/wiki/Ordered_words
# http://www.puzzlers.org/pub/wordlists/unixdict.txt

def ordered_word(w)
  letters = w.downcase.split(//).to_a
  (0...letters.count-1).each { |i| return false if letters[i] > letters[i+1] }
end

def find_ordered_words_in_dictionary(file="unixdict.txt",ordered_words=[])
  File.open(file).readlines.map { |w| w.chomp! }.each { |w| ordered_words << w if ordered_word(w) }
  ordered_words.group_by { |word| word.length }.max
end

puts find_ordered_words_in_dictionary

File: ordered_words.swift
// http://programmingpraxis.com/2015/07/14/ordered-words/

import Foundation

func orderedWord(word: String) -> Bool {
  return Array(word) == Array(word).sorted() { $0 < $1 }
}

func findOrderedWords(file: String) -> [String] {
  let fileContent = NSString(contentsOfFile: file, encoding: NSUTF8StringEncoding, error: nil)!
  let words = fileContent.componentsSeparatedByString("\n") as! [String]
  var orderedWords = [String]()
  for word in words {
    if orderedWord(word) {
      orderedWords.append(word)
    }
  }
  return orderedWords
}

func findLongestOrderedWord(words: [String]) -> String {
  var longestWord = ""
  var longestWordLength = 0
  for word in words {
    if count(word) > longestWordLength {
      longestWordLength = count(word)
      longestWord = word
    }
  }
  return longestWord
}

let orderedWordsFromUnixDict = findOrderedWords("unixdict.txt")
println(findLongestOrderedWord(orderedWordsFromUnixDict))
